src/main/java/com/hazelcast/gcp/GcpDiscoveryStrategyFactory.java

```diff
@@ -50,4 +50,47 @@ public DiscoveryStrategy newDiscoveryStrategy(DiscoveryNode discoveryNode, ILogg
         }
         return result;
     }
+
+    @Override
+    public boolean isAutoDetectionApplicable() {
+        return googleInternalDnsConfigured() && metadataFlavorGoogle();
+    }
+
+
+    private String readFileContents(String fileName) {
```

Comment:
Consider changing this method and others to `static`

Reply:
I changed it.